README.md

```diff
@@ -1,98 +1,169 @@
+    // If you return false, it will not be opened and the next step
+    // will be attempted to open.
+    print('The overlay is about to be displayed');
     return true;
   },
-  child: IconButton( // The actual widget that is in your interface
-    icon: icon,
-    onPressed: onPressed
-  ),
-);
-``` 
+```
 
-When you want to display the feature overlay, call:
+#### `enablePulsingAnimation`
 
+This is set to `true` by default, but you can disable the pulsing animation about the tap target by setting this to `false`.
 
-``` 
+#### `allowShowingDuplicate`
+
+If multiple `DescribedFeatureOverlay`s have the same `featureId`, they will interfere with each other during discovery and if you want to display multiple overlays at the same time, you will have to set `allowShowingDuplicate` to `true` for all of them.
+</details>
+
+### `FeatureDiscovery.discoverFeatures`
+
+When you want to showcase your features, you can call `FeatureDiscovery.discoverFeatures` with the applicable feature ids. The features will be displayed as steps in order if the user does not dismiss them.  
+By tapping the tap target, the user will be sent on to the next step and by tapping outside of the overlay, the user will dismiss all queued steps.
+
+```dart
 FeatureDiscovery.discoverFeatures(
   context,
-  const {'featureId1'}, // ids of your different overlays
+  const <String>{ // Feature ids for every feature that you want to showcase in order.
+    'add_item_feature_id',
+  },
 );
 ```
 
-### Note 
+If you want to display Feature Discovery for a page right after it has been opened, you can use [`SchedulerBinding.addPostFrameCallback`] in the [`initState` method of your `StatefulWidget`](https://api.flutter.dev/flutter/widgets/State/initState.html):
 
-#### contentLocation :
-we use this property for placing the text in proper position when lib can't do it (because of text width and height measurement issue in flutter )
-  
-ContentOrientation.below: move text under the target
-ContentOrientation.above: move text above the target
-ContentOrientation.trivial: let lib decide
-    
-## Bonus 
-
-When you'r desired target is in scrollable content and is hidden when the feature discovery runs you can use ```EnsureVisible``` widget like that
-    
+```dart
+@override
+void initState() {
+  // ...
+  SchedulerBinding.instance.addPostFrameCallback((Duration duration) {
+    FeatureDiscovery.discoverFeatures(
+      context,
+      const <String>{ // Feature ids for every feature that you want to showcase in order.
+        'add_item_feature_id',
+      },
+    ); 
+  });
+  super.initState();
+}
 ```
-GlobalKey<EnsureVisibleState> ensureKey2 = GlobalKey<EnsureVisibleState>();
+
+#### Other methods
+
+You can view the [API reference for `FeatureDiscovery`](https://pub.dev/documentation/feature_discovery/latest/feature_discovery/FeatureDiscovery-class.html#static-methods) to find other useful methods for controlling the Feature Discovery process programmatically.
```

Comment:
Unfortunately documentation doesn't appear for static methods :(

Reply:
Well, there is none in the current version of the plugin (on Pub), but you are saying that it will never show?

Reply:
There is one!

Reply:
In your version, but not [here](https://github.com/ayalma/feature_discovery/blob/master/lib/src/feature_discovery.dart#L15), which is what I linked to.

Reply:
Ooooh… Well, merge it now! Hoping @ayalma the original author of the plugin will do the same…

Reply:
For my [`wakelock` package](https://pub.dev/documentation/wakelock/latest/wakelock/Wakelock-class.html#static-methods), there is documentation for the static methods.

Reply:
>Hoping @ayalma the original author of the plugin will do the same… 

We will see. Their status says "I am slow to respond", so we can wait I guess.